ros/processor/inventory_events_consumer.py

```diff
@@ -133,39 +135,32 @@ def host_create_update_events(self, msg):
     def process_system_details(self, msg):
         """ Store new system information (stale, stale_warning timestamp) and return internal DB id"""
         host = msg['host']
-        performance_record = get_performance_profile(msg['platform_metadata']['url'], host['account'])
         with app.app_context():
             try:
                 account = get_or_create(
                     db.session, RhAccount, 'account',
                     account=host['account']
                 )
-
-                system = get_or_create(
-                    db.session, System, 'inventory_id',
-                    account_id=account.id,
-                    inventory_id=host['id'],
-                    display_name=host['display_name'],
-                    fqdn=host['fqdn'],
-                    cloud_provider=host['system_profile']['cloud_provider'],
-                    instance_type=performance_record.get('instance_type'),
-                    stale_timestamp=host['stale_timestamp']
-                )
-
-                if len(performance_record) == 2 and ('total_cpus' and 'instance_type' in performance_record):
-                    performance_utilization = {'memory': 0, 'cpu': 0, 'io': 0}
-                else:
-                    performance_utilization = self._calculate_performance_utilization(
-                        performance_record, host
+                db_fields = {
+                    "account_id": account.id,
+                    "inventory_id": host['id'],
+                    "display_name": host['display_name'],
+                    "fqdn": host['fqdn'],
+                    "cloud_provider": host['system_profile']['cloud_provider'],
+                    "stale_timestamp": host['stale_timestamp']
+                }
+                if msg['type'] == 'created':
+                    performance_record = get_performance_profile(
+                        msg['platform_metadata']['url'],
+                        host['account']
```

Comment:
@saltgen  -  does that mean ROS backend is going to download archive twice? one to get just `instance_type` and second to get new metrics data.

Reply:
@kgaikwad Good point. Yes that is correct,

1. to get just instance_type (events consumer)
2. actual performance record/new metrics data (engine consumer) as per changes [here](https://github.com/RedHatInsights/ros-backend/pull/138)

I did realize this while making the changes, but did not want to change the code too much.

Shall I set the instance_type on the engine consumer as well, to avoid multiple network requests?

Reply:
> Shall I set the instance_type on the engine consumer as well, to avoid multiple network requests?

Yes, sounds good to me!

Reply:
1. Removed saving of instance_type here, in-place tests run fine.
2. Added the same on https://github.com/RedHatInsights/ros-backend/pull/138, unit-tested as well.

Changes pushed :+1: